### PR TITLE
Improving docstrings for UpgradableModule interface

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #5466

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/05-port/types/module.go [hunk 1]
```diff
@@ -107,8 +107,11 @@ type IBCModule interface {
 }
 
 // UpgradableModule defines the callbacks required to perform a channel upgrade.
+// Note: applications must ensure that state related to packet processing remains unmodified until the OnChanUpgradeOpen callback is executed.
+// This guarantees that in-flight packets are correctly flushed using the existing channel parameters.
 type UpgradableModule interface {
-	// OnChanUpgradeInit initializes the channel upgrade handshake.
+	// OnChanUpgradeInit enables additional custom logic to be executed when the channel upgrade is initialized.
+	// Note: in the case of crossing hellos, this callback may be executed on both chains.
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:114]:
```suggestion
	// OnChanUpgradeInit enables additional custom logic to be executed when the channel upgrade is initialized.
	// It must validate the proposed version, order, and connection hops. 
	// Note: in the case of crossing hellos, this callback may be executed on both chains.
```

File: modules/core/05-port/types/module.go [hunk 2]
```diff
@@ -117,7 +120,8 @@ type UpgradableModule interface {
 		version string,
 	) (string, error)
 
-	// OnChanUpgradeTry verifies the counterparty upgrade and sets the upgrade on TRY chain
+	// OnChanUpgradeTry enables additional custom logic to be executed in the ChannelUpgradeTry step of the
+	// channel upgrade handshake.
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:124]:
```suggestion
	// OnChanUpgradeTry enables additional custom logic to be executed in the ChannelUpgradeTry step of the
	// channel upgrade handshake. It must validate the proposed version (provided by the counterparty), order, 
	// and connection hops. 
```

File: modules/core/05-port/types/module.go [hunk 3]
```diff
@@ -126,15 +130,17 @@ type UpgradableModule interface {
 		counterpartyVersion string,
 	) (string, error)
 
-	// OnChanUpgradeAck TODO
+	// OnChanUpgradeAck enables additional custom logic to be executed in the ChannelUpgradeAck step of the
+	// channel upgrade handshake.
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:134]:
```suggestion
	// OnChanUpgradeAck enables additional custom logic to be executed in the ChannelUpgradeAck step of the
	// channel upgrade handshake. It must validate the version proposed by the counterparty.
```

File: modules/core/05-port/types/module.go [hunk 4]
```diff
@@ -126,15 +130,17 @@ type UpgradableModule interface {
...
 	OnChanUpgradeAck(
 		ctx sdk.Context,
 		portID,
 		channelID,
 		counterpartyVersion string,
 	) error
 
-	// OnChanUpgradeOpen TODO
+	// OnChanUpgradeOpen enables additional custom logic to be executed when the channel upgrade has successfully completed, and the channel
+	// has returned to the OPEN state.
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:143]:
```suggestion
	// OnChanUpgradeOpen enables additional custom logic to be executed when the channel upgrade has successfully completed, and the channel
	// has returned to the OPEN state. Any logic associated with changing of the channel fields should be performed 
	// in this callback. 
```

File: modules/core/05-port/types/module.go [hunk 5]
```diff
@@ -144,7 +150,10 @@ type UpgradableModule interface {
 		version string,
 	)
 
-	// OnChanUpgradeRestore TODO
+	// OnChanUpgradeRestore enables additional custom logic to be executed when any of the following occur:
+	//	- the channel upgrade is aborted.
+	//	- the channel upgrade is cancelled.
+	//  - the channel upgrade times out.
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:156]: formatting?

Comment (colin-axner) [inline modules/core/05-port/types/module.go:156]:
```suggestion
	// OnChanUpgradeRestore enables additional custom logic to be executed when any of the following occur:
	//	- the channel upgrade is aborted.
	//	- the channel upgrade is cancelled.
	//  - the channel upgrade times out.
	// Any logic set due to an upgrade attempt should be reverted in this callback. 
```

**Review by AdityaSripal — APPROVED** [review]

**Review by charleenfei — APPROVED** [review]

**Review by crodriguezvega — APPROVED** [review]
Crystal clear documentation!

**Pull request merged**